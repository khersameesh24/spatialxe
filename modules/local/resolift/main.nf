process RESOLIFT {
    tag "$meta.id"
    label 'process_medium'

    container "quay.io/khersameesh24/resolift:1.0.0"

    input:
    tuple val(meta), path(input)

    output:
    tuple val(meta), path("*.tiff"), emit: enhanced_tiff
    path("versions.yml")           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "ResoLift module does not support Conda. Please use Docker / Singularity / Podman instead."
    }
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    if ("$input" == "${prefix}.tiff") error "Input and output names are the same, set prefix in module configuration to disambiguate!"
    def VERSION = '1.0.0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.

    """
    resolift \\
        -i $input \\
        -o ${prefix}.tiff \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        resolift: ${VERSION}
    END_VERSIONS
    """

    stub:
    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "ResoLift module does not support Conda. Please use Docker / Singularity / Podman instead."
    }
    def prefix = task.ext.prefix ?: "${meta.id}"
    if ("$input" == "${prefix}.tiff") error "Input and output names are the same, set prefix in module configuration to disambiguate!"
    def VERSION = '1.0.0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.

    """
    touch ${prefix}.tiff

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        resolift: ${VERSION}
    END_VERSIONS
    """
}
