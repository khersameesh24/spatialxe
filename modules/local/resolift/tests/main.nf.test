nextflow_process {

    name "Test Process RESOLIFT"
    script "../main.nf"
    process "RESOLIFT"

    tag "modules"
    tag "modules_local"
    tag "resolift"
    tag "segmentation"
    tag "cell_segmentation"


    test("resolift") {

        when {
            process {
                """
                input[0] = Channel.of([
                            [id: "test_run"], ["/workspace/spatialxe/modules/local/resolift/tests/morphology.ome.tif"]])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }

    }

    test("resolift stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                            [id: "test_run"], ["/workspace/spatialxe/modules/local/resolift/tests/morphology.ome.tif"]])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
