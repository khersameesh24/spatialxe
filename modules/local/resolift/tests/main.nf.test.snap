{
    "resolift stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_run"
                        },
                        "test_run.tiff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e2a649848a8b90aecf354b1478c05450"
                ],
                "output": [
                    [
                        {
                            "id": "test_run"
                        },
                        "test_run.tiff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2a649848a8b90aecf354b1478c05450"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-25T15:17:43.654819314"
    },
    "resolift main": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_run"
                        },
                        "test_run.tiff:md5,45735fdfe69f69e6fb9556c317c17a7c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e2a649848a8b90aecf354b1478c05450"
                ],
                "output": [
                    [
                        {
                            "id": "test_run"
                        },
                        "test_run.tiff:md5,45735fdfe69f69e6fb9556c317c17a7c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2a649848a8b90aecf354b1478c05450"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-25T15:30:32.695757643"
    }
}